src/url-params.ts

```diff
@@ -175,12 +175,18 @@ export class AnalyticsUrlParams {
 	/**
 	 * @return full query parameter string that can be appended to URLs
 	 */
-	getQueryString(destinationUrl?: URL, currentUrl?: URL): string {
+	getQueryString(destinationUrl?: URL | string, currentUrl?: URL): string {
+		// we first check if destionationUrl is a relative URL string. If it is, we exit.
+		//const relativeRegex = /^(?!www\.|(?:http|ftp)s?:\/\/|[A-Za-z]:\\|\/\/).*/g;
+		let destinationDomainMatch;
+		let destinationDomain;
+		let actualDomainMatch;
+		let actualDomain;
```

Comment:
All these are typed as `any`, which isn't good...
After the changes suggested bellow this will be:
```suggestion
		let destinationDomain: string | undefined;
```
but I would suggest to push it lower, near the place it's first assinged.